ui/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.js

```diff
@@ -45,17 +48,44 @@ function getMatchTypeAttribute() {
     : 'datatype'
 }
 
-function isTypeLimiter(filter) {
-  let typesFound = {}
-  filter.filters.forEach(subfilter => {
-    typesFound[CQLUtils.getProperty(subfilter)] = true
+const getMatchTypes = memoize(async () => {
+  const matchTypeAttr = getMatchTypeAttribute()
+  const json = await query({
+    count: 0,
+    cql: "anyText ILIKE '*'",
+    facets: [matchTypeAttr],
   })
-  typesFound = Object.keys(typesFound)
-  return (
-    typesFound.length === 2 &&
-    typesFound.indexOf('metadata-content-type') >= 0 &&
-    typesFound.indexOf(getMatchTypeAttribute()) >= 0
-  )
+  const facets = json.facets[matchTypeAttr] || []
+  return facets
+    .map(facet => facet.value)
+    .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()))
+    .map(value => ({
+      label: value,
+      value,
+      class: 'icon ' + IconHelper.getClassByName(value),
+    }))
+})
+
+const NoMatchTypesView = Marionette.ItemView.extend({
+  template() {
+    return `No types found for '${getMatchTypeAttribute()}'.`
+  },
+})
+
+function isTypeLimiter(filter) {
+  const typesFound = _.uniq(filter.filters.map(CQLUtils.getProperty))
+  const metadataContentTypeSupported = !!metacardDefinitions.metacardTypes[
```

Comment:
Do you need the `!!`?

Reply:
No but I made it a boolean to make it clear I am using the result solely in conditional statements.

Reply:
:pencil2: I think it would read better to create new boolean instead
```suggestion
  const metadataContentTypeSupported = Boolean(metacardDefinitions.metacardTypes[METADATA_CONTENT_TYPE])
```

Reply:
Done